lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -201,16 +201,33 @@ def connect
 
         connection = MarchHare.connect(rabbitmq_settings)
 
-
-        connection.on_blocked { @logger.warn("RabbitMQ output blocked! Check your RabbitMQ instance!") }
-        connection.on_unblocked { @logger.warn("RabbitMQ output unblocked!") }
+        connection.on_shutdown do |conn, cause|
+           @logger.warn("RabbitMQ connection was closed!",
+                          :url => connection_url(conn),
+                          :automatic_recovery => @automatic_recovery,
+                          :cause => cause)
+        end
+        connection.on_blocked do
+          @logger.warn("RabbitMQ connection blocked! Check your RabbitMQ instance!",
+                       :url => connection_url(connection))
+        end
+        connection.on_unblocked do
+          @logger.warn("RabbitMQ connection unblocked!", :url => connection_url(connection))
```

Comment:
This has been warn for a long time. I think it is something that is warning level, normally queues should not be blocked unless there's a problem.